c/authService.c

```diff
@@ -173,6 +174,202 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "No new password for expired password provided");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_WRONG_FORMAT:
+          setResponseStatus(response,HTTP_STATUS_BAD_REQUEST,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "The new password format is incorrect. Please try again.");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_TOO_MANY_ATTEMPTS:
+          setResponseStatus(response,HTTP_STATUS_TOO_MANY_REQUESTS,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "An incorrect password has been entered too many times and the account has been locked. Please contact your administrator for further instructions.");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        default:
+          setResponseStatus(response,HTTP_STATUS_BAD_REQUEST,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          char responseString[RESPONSE_MESSAGE_LENGTH];
+          sprintf(responseString, "Password reset FAILED with return code: %d reason code: %d\n", returnCode, reasonCode);
+
+          jsonStart(p);
+		  jsonAddString(p, "status", responseString);
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+      }
+    }
+  } else {
+    setResponseStatus(response,HTTP_STATUS_METHOD_NOT_FOUND,"ERROR");
+    setContentType(response,"text/json");
+    addStringHeader(response,"Server","jdmfws");
+    addStringHeader(response,"Transfer-Encoding","chunked");
+    addStringHeader(response, "Cache-control", "no-store");
+    addStringHeader(response, "Pragma", "no-cache");
+    writeHeader(response);
+    jsonStart(p);
+    jsonAddString(p, "status", "Method Not Allowed");
```

Comment:
Also did you know, if you are just doing a start-string-end json, you can do: https://github.com/zowe/zowe-common-c/blob/master/c/httpserver.c#L4019